pkg/storage/store_pool.go

```diff
@@ -327,6 +327,20 @@ func (sp *StorePool) deadReplicasGossipUpdate(_ string, content roachpb.Value) {
 	detail.deadReplicas = deadReplicas
 }
 
+// updateLocalStoreAfterRebalance is used to update local copy of target store
+// after we make an rebalance immediately.
+func (sp *StorePool) updateLocalStoreAfterRebalance(storeID roachpb.StoreID, repl *Replica) {
+	sp.detailsMu.Lock()
+	defer sp.detailsMu.Unlock()
+	detail := sp.getStoreDetailLocked(storeID)
```

Comment:
I think you'll have to make a copy here and reinsert the copy into the `StorePool` rather than modifying the returned detail in place, since there could be other parts of the code accessing the detail without`sp.detailsMu` held.

Reply:
So we should do it like: `detail := *sp.getStoreDetailLocked(storeID)`, then we update the `detail` and reinsert it into  the `StorePool`?

Reply:
Yes, something with those semantics. I'm worried it may create data races to modify the `detail` object in place.

Reply:
Done.